[cutlass backend] Reduce log level for cutlass runtime error

henrylhtsang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #153765
* __->__ #153457

Want to make sure we always call self.cleanup_run_fn() even if we crash.

I think this is the reason why sometimes we get 
```
in _dlclose
TypeError: 'NoneType' object is not callable
```

Differential Revision: [D74629230](https://our.internmc.facebook.com/intern/diff/D74629230/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang committed:
[cutlass backend] Reduce log level for cutlass runtime error



Differential Revision: [D74629230](https://our.internmc.facebook.com/intern/diff/D74629230/)

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] Reduce log level for cutlass runtime error"


Want to make sure we always call self.cleanup_run_fn() even if we crash.

I think this is the reason why sometimes we get 
```
in _dlclose
TypeError: 'NoneType' object is not callable
```

Differential Revision: [D74629230](https://our.internmc.facebook.com/intern/diff/D74629230/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler commented on torch/_inductor/select_algorithm.py:
```diff
@@ -2265,16 +2265,30 @@ def benchmark_choices(
+
+                if not isinstance(choice, CUDATemplateCaller):
+                    log.error(
+                        "CUDA compilation error during autotuning: \n%s. \nIgnoring this choice.",
```
is this compilation error? or runtime error?

ColinPeppler commented on torch/_inductor/autotune_process.py:
```diff
@@ -727,6 +727,20 @@ def make_run_fn(
+        except RuntimeError as e:
+            err_msg = str(e)
+
+            def dummy_function():
+                raise RuntimeError(err_msg)
```
for readability, change the fn name to be more descriptive!!!

maybe `raise_runtime_error()`

ColinPeppler commented on torch/_inductor/select_algorithm.py:
```diff
@@ -2265,16 +2265,30 @@ def benchmark_choices(
+                        msg,
+                        exc_info=True,
+                    )
                 else:
-                    if "illegal memory access" in msg:
-                        msg += "\n\nEither error in template or triton bug.\n"
-                log.error(
-                    "Runtime error during autotuning: \n%s. \nIgnoring this choice.",
-                    msg,
-                )
+                    log.error(
+                        "Runtime error during autotuning: \n%s. \nIgnoring this choice.",
+                        msg,
+                    )
```
it'd be easier to read if we just log this once, but change the log-level and exc_info depending on if it came from CUDATemplateCaller or not.

henrylhtsang replied:
> it'd be easier to read if we just log this once, but change the log-level and exc_info depending on if it came from CUDATemplateCaller or not.

I thought about it a few times, don't tink there is a easy to to do

henrylhtsang committed:
Update on "[cutlass backend] Reduce log level for cutlass runtime error"


Want to make sure we always call self.cleanup_run_fn() even if we crash.

I think this is the reason why sometimes we get 
```
in _dlclose
TypeError: 'NoneType' object is not callable
```

Differential Revision: [D74629230](https://our.internmc.facebook.com/intern/diff/D74629230/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler commented on torch/_inductor/select_algorithm.py:
```diff
@@ -2265,16 +2265,31 @@ def benchmark_choices(
             except RuntimeError as e:
+                from torch._inductor.codegen.cuda.cuda_kernel import CUDATemplateCaller
+
+                if not isinstance(choice, CUDATemplateCaller):
+                    log.error(
+                        "CUDA runtime error during autotuning: \n%s. \nIgnoring this choice.",
+                        e,
+                    )
```
is it possible this could be a ROCM runtime error?

ColinPeppler replied:
Also, I see there's another log.error below for the non-CUDATemplateCaller path. Should we just log this error once?

henrylhtsang committed:
Update on "[cutlass backend] Reduce log level for cutlass runtime error"


Want to make sure we always call self.cleanup_run_fn() even if we crash.

I think this is the reason why sometimes we get 
```
in _dlclose
TypeError: 'NoneType' object is not callable
```

Differential Revision: [D74629230](https://our.internmc.facebook.com/intern/diff/D74629230/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] Reduce log level for cutlass runtime error"


Want to make sure we always call self.cleanup_run_fn() even if we crash.

I think this is the reason why sometimes we get 
```
in _dlclose
TypeError: 'NoneType' object is not callable
```

Differential Revision: [D74629230](https://our.internmc.facebook.com/intern/diff/D74629230/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] Reduce log level for cutlass runtime error"


Want to make sure we always call self.cleanup_run_fn() even if we crash.

I think this is the reason why sometimes we get 
```
in _dlclose
TypeError: 'NoneType' object is not callable
```

Differential Revision: [D74629230](https://our.internmc.facebook.com/intern/diff/D74629230/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
@pytorchbot merge -i
